horology/tformatter.py

```diff
@@ -1,37 +1,57 @@
 
-UNITS = {
-    'ns': 10 ** -9,
-    'us': 10 ** -6,
-    'ms': 10 ** -3,
-    's': 1,
-    'min': 60,
-    'h': 3600,
-    'd': 3600 * 24
-}  # todo: allow alternatives, eg. h == hs == hours
-
-
-def rescale_time(t, unit):
+# A tuple of tuples containing:
+# ( <unit_names>, <rescale_calc>, <auto_cal> )
+
+UNITS = (
+    (('ns'), (10 ** -9), (10 ** -6)),
```

Comment:
Use [namedtuple](https://docs.python.org/3.8/library/collections.html#collections.namedtuple).

Reply:
Yes, I forgot to add a comma to the tuples that are single so it was finding the first `letter` and not the first element!

Furthermore the return `d` was in the loop so that was that...:P

I did not choose namedtuple because it is not needed.  Only the first element needs to be a tuple to add possible variants of the measurements, I removed the calculations out of tuples and kept them as a int/floats only. :)

I have fixed these and all tests now pass so I will  re-commit something :)

Reply:
oops ^^ that is me but from my work account.. sorry about that..  :)

Reply:
> Dear @johnashu I am very happy to see your commits!
> 
> Please make sure that all tests pass. You can run them with:
> 
> ```
> nosetests -vv --with-doctest --doctest-options=+ELLIPSIS
> ```
> 
> Probably you need to install nose before. It can be done with pip or conda:
> 
> ```
> pip install nose
> ```
> 
> I run tests on your branch and unfortunately 8 tests failed.
> 
> In the meantime I updated unit tests on master and I plan to add CI that will run them on every commit and PR. But for now make sure that you executed tests locally on your machine.

I have changed the code and tests passed :)

Reply:
> I did not choose namedtuple because it is not needed. 

That being said, it may help with readability if you wish?

Reply:
Yes, please. I mean this precisly:
```python
Unit = namedtuple('Unit', 'names scale limit')
```